src/lua/backtrace.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include "lua/backtrace.h"
+
+#ifdef ENABLE_BACKTRACE
+#include "core/fiber.h"
+#include "core/tt_static.h"
+
+#include "lua.h"
+
+/*
+ * This is the declaration of the Lua stack entry function: it does not exactly
+ * `lj_BC_FUNCC`'s signature and its only aim is to provide the `lj_BC_FUNCC`
+ * symbol's address for setting up Lua stack boundaries.
+ */
+void
+lj_BC_FUNCC(void);
+
+static void *lua_stack_entry_start_ip = (void *)lj_BC_FUNCC;
+/* See lj_vm.s */
+static void *lua_stack_entry_end_ip = (char *)lj_BC_FUNCC + 72;
+
+static const char *const lua_frame_str_fmt = "#%-2d %s in %s:%d";
+
+void
+backtrace_lua_init(void)
+{
+#ifndef __APPLE__
+	unw_proc_info_t proc_info;
+	int rc = unw_get_proc_info_by_ip(unw_local_addr_space,
+					 (unw_word_t)lua_stack_entry_start_ip,
+					 &proc_info, NULL);
+	if (rc != 0) {
+		say_debug("unwinding error: unw_get_proc_info_by_ip failed: "
+			  "%s", unw_strerror(rc));
+		return;
+	}
+	lua_stack_entry_start_ip = (void *)proc_info.start_ip;
+	lua_stack_entry_end_ip = (void *)proc_info.end_ip;
+#endif /* __APPLE__ */
+}
+
+/*
+ * Append frame to backtrace.
+ */
+static void
+append_frame(struct backtrace_lua *bt, struct backtrace_frame *c_frame)
```

Comment:
Nit: `append_c_frame` would be a better name, otherwise one might think that `append_lua_frame` calls `append_frame`.

Reply:
Agree, sounds reasonable.

Fixed: renamed `append_frame` to distinguish the frame type.